pyro/contrib/gp/__init__.py

```diff
@@ -1,15 +1,136 @@
 from __future__ import absolute_import, division, print_function
 
+from torch.autograd import Variable
 import torch.nn as nn
-from torch.nn import Parameter
+from torch.distributions import transform_to
 
+import pyro
+import pyro.distributions as dist
 
-class InducingPoints(nn.Module):
 
-    def __init__(self, Xu, name="inducing_points"):
-        super(InducingPoints, self).__init__()
-        self.inducing_points = Parameter(Xu)
-        self.name = name
+class Parameterized(nn.Module):
+    """
+    Parameterized class.
 
-    def forward(self):
-        return self.inducing_points
+    This is a base class for other classes in Gaussian Process.
+    By default, a parameter will be a ``torch.nn.Parameter`` containing ``torch.FloatTensor``.
+    To cast them to the correct data type or GPU device, we can call methods such as
+    ``.double()``, ``.cuda(device=0)``,...
+    See `torch.nn.Module
+    <http://pytorch.org/docs/master/nn.html#torch.nn.Module>`_ for more information.
+    """
+
+    def __init__(self):
+        super(Parameterized, self).__init__()
+        self._priors = {}
+        self._constraints = {}
+        self._name = None
+        self._registered_params = {}
+        self._fixed_params = {}
+
+    def set_prior(self, param, prior):
+        """
+        Sets a prior to a parameter. Use `set_prior(param, None)` to remove the prior distribution of
```

Comment:
Is there a use case for unsetting a prior? If there's no foreseeable use case, it would be better to not support this functionality.

Reply:
The use case is when we want to reuse some modules, but are lazy to redefine it. ^^! Because it is not important (in my case, I don't use it), I will delete that part of docs.